src/detect-flowbits.c

```diff
@@ -70,6 +74,40 @@ void DetectFlowbitsRegister (void)
     DetectSetupParseRegexes(PARSE_REGEX, &parse_regex, &parse_regex_study);
 }
 
+static char **StrSplit(char *arrptr, const char *delim)
+{
+    char **strarr = NULL;
+    char *token, *saveptr;
+    uint8_t i = 0;
+    if (strchr(arrptr, ' ') != NULL) {
+        SCLogError(SC_ERR_INVALID_SIGNATURE, "Space not allowed in flowbits.");
+        goto error;
+    }
+    strarr = SCCalloc((MAX_TOKENS + 1), sizeof(char **));
+    if (strarr == NULL)
+        goto error;
+    while ((token = strtok_r(arrptr, delim, &saveptr))) {
+        if (i >= MAX_TOKENS) {
```

Comment:
`i` will never exceed `MAX_TOKENS`.

Can the check be moved to just before (or after) the `i++`?

This changes things slightly -- token collection ceases when the maximum number of tokens are collected instead of generating an error when the maximum is reached and there are more tokens to collect.

Reply:
I think I am misunderstanding something here. I changed MAX_`TOKENS` to 2 and passed three flowbits in a test and it failed with the error message here.
Could you please elaborate a little more which situation would `i` not exceed `MAX_TOKENS`?